src/Mvc/Mvc.Razor/src/ApplicationParts/RazorCompiledItemFeatureProvider.cs

```diff
@@ -36,10 +35,49 @@ public void PopulateFeature(IEnumerable<ApplicationPart> parts, ViewsFeature fea
 
                 foreach (var item in provider.CompiledItems)
                 {
-                    var descriptor = new CompiledViewDescriptor(item);
+                    var compiledItem = item;
+                    if (_hotReloadedViews is not null && _hotReloadedViews.TryGetValue(item.Identifier, out var hotReloadedType))
+                    {
+                        // Determine if a hot reload update is available for this view.
+                        compiledItem = new HotReloadRazorCompiledItem(item, hotReloadedType);
+                    }
+
+                    var descriptor = new CompiledViewDescriptor(compiledItem);
                     feature.ViewDescriptors.Add(descriptor);
                 }
             }
         }
+
+        public void UpdateCache(Type[]? types)
+        {
+            if (types is null)
+            {
+                return;
+            }
+
+            foreach (var type in types)
+            {
+                if (type.GetCustomAttribute<RazorFileIdentifierAttribute>() is { } attribute)
```

Comment:
```suggestion
                if (type.GetCustomAttribute<RazorFileIdentifierAttribute>() is RazorFileIdentifierAttribute attribute)
```

Nit: but it's a little hard to grok the condition when `{ }` is used without a pattern to match against.